pkg/openstack/loadbalancer.go

```diff
@@ -515,10 +516,34 @@ func (lbaas *LbaasV2) createOctaviaLoadBalancer(service *corev1.Service, name, c
 		createOpts.VipAddress = loadBalancerIP
 	}
 
+	for _, port := range service.Spec.Ports {
+		listenerCreateOpt := lbaas.buildListenerCreateOpt(port, svcConf)
+		members, newMembers, err := lbaas.buildBatchUpdateMemberOpts(port, nodes, svcConf)
+		if err != nil {
+			return nil, err
+		}
+		poolCreateOpt := lbaas.buildPoolCreateOpt(string(listenerCreateOpt.Protocol), service, svcConf)
+		poolCreateOpt.Members = members
+		var withHealthMonitor string
+		if svcConf.enableMonitor {
+			opts := lbaas.buildMonitorCreateOpts(port)
+			poolCreateOpt.Monitor = &opts
+			withHealthMonitor = " with healthmonitor"
+		}
+
+		listenerCreateOpt.DefaultPool = &poolCreateOpt
+		createOpts.Listeners = append(createOpts.Listeners, listenerCreateOpt)
+		klog.V(2).Infof("Loadbalancer %s: adding pool%s using protocol %s with %d members", name, withHealthMonitor, poolCreateOpt.Protocol, len(newMembers))
+	}
+
 	mc := metrics.NewMetricContext("loadbalancer", "create")
 	loadbalancer, err := loadbalancers.Create(lbaas.lb, createOpts).Extract()
 	if mc.ObserveRequest(err) != nil {
-		return nil, fmt.Errorf("error creating loadbalancer %v: %v", createOpts, err)
+		var printObj interface{} = createOpts
+		if opts, err := json.Marshal(createOpts); err == nil {
+			printObj = string(opts)
+		}
+		return nil, fmt.Errorf("error creating loadbalancer %v: %s", printObj, err)
```

Comment:
we better use %v , to print the error here?

Reply:
thanks, will adapt it.